src/ray/object_manager/test/object_manager_stress_test.cc

```diff
@@ -0,0 +1,395 @@
+    status =
+        server1->object_manager_.SubscribeObjAdded([this](const ObjectID &object_id) {
+          object_added_handler_1(object_id);
+          if (v1.size() == num_expected_objects && v1.size() == v2.size()) {
+            TransferTestComplete();
+          }
+        });
+    RAY_CHECK_OK(status);
+    status =
+        server2->object_manager_.SubscribeObjAdded([this](const ObjectID &object_id) {
+          object_added_handler_2(object_id);
+          if (v2.size() == num_expected_objects && v1.size() == v2.size()) {
+            TransferTestComplete();
+          }
+        });
+    RAY_CHECK_OK(status);
+  }
+
+  void TransferTestNext() {
+    async_loop_index += 1;
+    if ((uint)async_loop_index < async_loop_patterns.size()) {
+      TransferPattern pattern = async_loop_patterns[async_loop_index];
+      TransferTestExecute(1000, 100, pattern);
+    } else {
+      main_service.stop();
+    }
+  }
+
+  plasma::ObjectBuffer GetObject(plasma::PlasmaClient &client, ObjectID object_id) {
+    plasma::ObjectBuffer object_buffer;
+    plasma::ObjectID plasma_id = object_id.to_plasma_id();
+    ARROW_CHECK_OK(client.Get(&plasma_id, 1, 0, &object_buffer));
+    return object_buffer;
+  }
+
+  static unsigned char *GetDigest(plasma::PlasmaClient &client, ObjectID object_id) {
+    const int64_t size = sizeof(uint64_t);
+    static unsigned char digest_1[size];
+    ARROW_CHECK_OK(client.Hash(object_id.to_plasma_id(), &digest_1[0]));
+    return digest_1;
+  }
+
+  void CompareObjects(ObjectID object_id_1, ObjectID object_id_2) {
+    plasma::ObjectBuffer object_buffer_1 = GetObject(client1, object_id_1);
+    plasma::ObjectBuffer object_buffer_2 = GetObject(client1, object_id_1);
+    uint8_t *data_1 = const_cast<uint8_t *>(object_buffer_1.data->data());
+    uint8_t *data_2 = const_cast<uint8_t *>(object_buffer_2.data->data());
+    ASSERT_EQ(object_buffer_1.data_size, object_buffer_2.data_size);
+    for (int i = -1; ++i < object_buffer_1.data_size;) {
+      ASSERT_TRUE(data_1[i] == data_2[i]);
+    }
+  }
+
+  void CompareHashes(ObjectID object_id_1, ObjectID object_id_2) {
+    const int64_t size = sizeof(uint64_t);
+    static unsigned char *digest_1 = GetDigest(client1, object_id_1);
+    static unsigned char *digest_2 = GetDigest(client2, object_id_2);
+    for (int i = -1; ++i < size;) {
+      ASSERT_TRUE(digest_1[i] == digest_2[i]);
+      if (digest_1[i] != digest_2[i]) {
+        RAY_LOG(INFO) << i << " " << digest_1[i] << " " << digest_2[i];
+      }
+    }
+  }
+
+  void TransferTestComplete() {
+    RAY_LOG(INFO) << "TransferTestComplete: " << async_loop_patterns[async_loop_index]
+                  << " " << v1.size() << " " << v2.size();
+    ASSERT_TRUE(v1.size() == v2.size());
+    for (int i = -1; ++i < (int)v1.size();) {
+      ASSERT_TRUE(std::find(v1.begin(), v1.end(), v2[i]) != v1.end());
+    }
+
+    // Compare objects and their hashes.
+    for (int i = -1; ++i < (int)v1.size();) {
```

Comment:
ah, this is an instance of comparing signed and unsigned integers. I would rather `int -> uint` here than typecast `size()`